engine/src/main/java/org/terasology/engine/logic/console/ui/NotificationOverlay.java

```diff
@@ -75,6 +82,18 @@ public String get() {
                 return messageHistory.toString();
             }
         });
+
+        // hide the chat notification overlay when the rest of the HUD is hidden
+        //FIXME? To bind the 'visible' property which already exists on this screen layer we need to create
+        //       a new binding. To avoid unnecessary lookup calls we use a default binding which we modify
+        //       imperatively based on the property change events we receive from the rendering config.
+        Binding<Boolean> visible = new DefaultBinding<>(true);
+        this.bindVisible(visible);
+        //FIXME? The debug rendering config is properly informed about hiding the HUD, whereas the
+        //       corresponding method on the NUI manager 'NUIManager#isHUDVisible' does not work properly.
+        config.getRendering().getDebug().subscribe(RenderingDebugConfig.HUD_HIDDEN, evt -> {
+            visible.set(!(Boolean) evt.getNewValue());
+        });
```

Comment:
confusing that this is from some kind of debug?

Reply:
not really - the only system reacting on that key press is actually the debug system, where "debug system" is a system handling different in-game options for debugging, like rendering chunk bounding boxes.

It's just not propagated anywhere else, and I'm not really sure what the NUI manager property is supposed to express.